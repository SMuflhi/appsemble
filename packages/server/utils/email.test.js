import {
  processTemplate,
  resendVerificationEmail,
  sendEmail,
  sendOrganizationInviteEmail,
  sendResetPasswordEmail,
  sendWelcomeEmail,
} from './email';

describe('sendMail', () => {
  it('should convert markdown to text and html', async () => {
    const result = await sendEmail(
      { to: 'test@example.com', from: 'test@example.com', subject: 'Test' },
      '**Bold Text** Regular text',
      null,
    );

    const converted = JSON.parse(result.message);

    expect(converted.to).toMatchSnapshot();
    expect(converted.html).toMatchSnapshot();
    expect(converted.markdown).toMatchSnapshot();
  });
});

describe('processTemplate', () => {
  it('should insert template variables', async () => {
    const template = 'Hello **<%= name %>**';
    const { attributes, content } = processTemplate(template, { name: 'John Doe' });

    expect(attributes).toStrictEqual({ name: 'John Doe' });
    expect(content).toBe('Hello **John Doe**');
  });

  it('should extract template variables', async () => {
    const template = '---\nsubject: Test\n---\nTest Message';
    const { attributes, content } = processTemplate(template, {});

    expect(attributes).toStrictEqual({ subject: 'Test' });
    expect(content).toBe('Test Message');
  });
});

describe('sendWelcomeEmail', () => {
  it('should match its snapshot', async () => {
    const result = await sendWelcomeEmail(
      { email: 'test@example.com', name: 'John Doe', url: 'https://example.com/test' },
      null,
    );

    const converted = JSON.parse(result.message);
    converted.messageId = '<TestMessageId>';
    expect(converted).toMatchSnapshot();
  });
});

describe('resendVerificationEmail', () => {
  it('should match its snapshot', async () => {
    const result = await resendVerificationEmail(
      { email: 'test@example.com', name: 'John Doe', url: 'https://example.com/test' },
      null,
    );

    const converted = JSON.parse(result.message);
    converted.messageId = '<TestMessageId>';
    expect(converted).toMatchSnapshot();
  });
});

describe('sendForgetPasswordEmail', () => {
  it('should match its snapshot', async () => {
    const result = await sendResetPasswordEmail(
      { email: 'test@example.com', name: 'John Doe', url: 'https://example.com/test' },
      null,
    );

    const converted = JSON.parse(result.message);
    converted.messageId = '<TestMessageId>';
    expect(converted).toMatchSnapshot();
  });
});

describe('sendOrganizationInviteEmail', () => {
  it('should match its snapshot', async () => {
    const result = await sendOrganizationInviteEmail(
      { email: 'test@example.com', organization: 'Appsemble', url: 'https://example.com/test' },
      null,
    );

    const converted = JSON.parse(result.message);
    converted.messageId = '<TestMessageId>';
    expect(converted).toMatchSnapshot();
  });
});
