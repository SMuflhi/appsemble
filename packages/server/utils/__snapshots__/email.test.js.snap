// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resendVerificationEmail should match its snapshot 1`] = `
Object {
  "bcc": null,
  "cc": null,
  "from": null,
  "headers": Object {},
  "html": "<p>Hello John Doe,</p>
<p>Someone has requested to resend the verification key for your account.<br>Before you can use your account, we need to verify your email address.</p>
<p>Please follow the link below to verify your email address:<br><a href=\\"https://example.com/test\\">https://example.com/test</a></p>
<p><strong>Didn’t request this email?</strong><br>No worries! Your address may have been entered by mistake. If you ignore or delete this email,
nothing will happen to your account.</p>
<p>Kind regards,</p>
<p><em>The Appsemble Team</em></p>
",
  "markdown": "Hello John Doe,

Someone has requested to resend the verification key for your account.  
Before you can use your account, we need to verify your email address.

Please follow the link below to verify your email address:  
https://example.com/test

**Didn’t request this email?**  
No worries! Your address may have been entered by mistake. If you ignore or delete this email,
nothing will happen to your account.

Kind regards,

_The Appsemble Team_
",
  "messageId": "<TestMessageId>",
  "subject": "Confirm account registration",
  "text": "Hello John Doe,

Someone has requested to resend the verification key for your account.  
Before you can use your account, we need to verify your email address.

Please follow the link below to verify your email address:  
https://example.com/test

**Didn’t request this email?**  
No worries! Your address may have been entered by mistake. If you ignore or delete this email,
nothing will happen to your account.

Kind regards,

_The Appsemble Team_
",
  "to": Array [
    Object {
      "address": "test@example.com",
      "name": "John Doe",
    },
  ],
}
`;

exports[`sendForgetPasswordEmail should match its snapshot 1`] = `
Object {
  "bcc": null,
  "cc": null,
  "from": null,
  "headers": Object {},
  "html": "<p>Hello John Doe,</p>
<p>A password reset has been requested for you account. Please click <a href=\\"https://example.com/test\\">here</a> to reset your
password.</p>
<p>Kind regards,</p>
<p><em>The Appsemble Team</em></p>
<p><strong>Didn’t request this email?</strong></p>
<p>Your email address may have been entered by mistake. This email can be ignored safely.</p>
",
  "markdown": "Hello John Doe,

A password reset has been requested for you account. Please click [here](https://example.com/test) to reset your
password.

Kind regards,

_The Appsemble Team_

**Didn’t request this email?**

Your email address may have been entered by mistake. This email can be ignored safely.
",
  "messageId": "<TestMessageId>",
  "subject": "Reset password",
  "text": "Hello John Doe,

A password reset has been requested for you account. Please click [here](https://example.com/test) to reset your
password.

Kind regards,

_The Appsemble Team_

**Didn’t request this email?**

Your email address may have been entered by mistake. This email can be ignored safely.
",
  "to": Array [
    Object {
      "address": "test@example.com",
      "name": "John Doe",
    },
  ],
}
`;

exports[`sendMail should convert markdown to text and html 1`] = `
Array [
  Object {
    "address": "test@example.com",
    "name": "",
  },
]
`;

exports[`sendMail should convert markdown to text and html 2`] = `
"<p><strong>Bold Text</strong> Regular text</p>
"
`;

exports[`sendMail should convert markdown to text and html 3`] = `"**Bold Text** Regular text"`;

exports[`sendOrganizationInviteEmail should match its snapshot 1`] = `
Object {
  "bcc": null,
  "cc": null,
  "from": null,
  "headers": Object {},
  "html": "<p>Hello,</p>
<p>You have been invited to join the organization <strong>Appsemble</strong>.</p>
<p>Joining an organization allows you to view private applications as well as create and edit apps
belonging to an organization.</p>
<p>You can accept or decline this invitation by following the link below:</p>
<p><a href=\\"https://example.com/test\\">https://example.com/test</a></p>
<p>Kind regards,</p>
<p><em>The Appsemble Team</em></p>
",
  "markdown": "Hello,

You have been invited to join the organization **Appsemble**.

Joining an organization allows you to view private applications as well as create and edit apps
belonging to an organization.

You can accept or decline this invitation by following the link below:

https://example.com/test

Kind regards,

_The Appsemble Team_
",
  "messageId": "<TestMessageId>",
  "subject": "Invited to organization",
  "text": "Hello,

You have been invited to join the organization **Appsemble**.

Joining an organization allows you to view private applications as well as create and edit apps
belonging to an organization.

You can accept or decline this invitation by following the link below:

https://example.com/test

Kind regards,

_The Appsemble Team_
",
  "to": Array [
    Object {
      "address": "test@example.com",
      "name": "",
    },
  ],
}
`;

exports[`sendWelcomeEmail should match its snapshot 1`] = `
Object {
  "bcc": null,
  "cc": null,
  "from": null,
  "headers": Object {},
  "html": "<p>Hello John Doe,</p>
<p>Thank you for registering your account.<br>Before you can use your account, we need to verify your email address.</p>
<p>Please follow the link below to verify your email address:<br><a href=\\"https://example.com/test\\">https://example.com/test</a></p>
<p>Kind regards,</p>
<p><em>The Appsemble Team</em></p>
",
  "markdown": "Hello John Doe,

Thank you for registering your account.  
Before you can use your account, we need to verify your email address.

Please follow the link below to verify your email address:  
https://example.com/test

Kind regards,

_The Appsemble Team_
",
  "messageId": "<TestMessageId>",
  "subject": "Welcome to Appsemble",
  "text": "Hello John Doe,

Thank you for registering your account.  
Before you can use your account, we need to verify your email address.

Please follow the link below to verify your email address:  
https://example.com/test

Kind regards,

_The Appsemble Team_
",
  "to": Array [
    Object {
      "address": "test@example.com",
      "name": "John Doe",
    },
  ],
}
`;
